mne/io/boxy/boxy.py

```diff
@@ -0,0 +1,664 @@
+                    boxy_coords[i_chan][:9].reshape(3, 3)).ravel()
+            else:
+                # Don't want to transform markers.
+                temp_ch_src_det = np.zeros(9,)
+            # Add wavelength and placeholders.
+            temp_other = np.asarray(boxy_coords[i_chan][9:], dtype=np.float64)
+            info['chs'][i_chan]['loc'] = np.concatenate((temp_ch_src_det,
+                                                        temp_other), axis=0)
+
+        raw_extras = {'source_num': source_num,
+                      'detect_num': detect_num,
+                      'start_line': start_line,
+                      'end_line': end_line,
+                      'filetype': filetype,
+                      'files': files,
+                      'montages': mtg_names,
+                      'blocks': blk_names,
+                      'data_types': data_types,
+                      'mtg_mdf': mtg_mdf,
+                      }
+
+        # Check data start lines.
+        if len(set(start_line)) == 1:
+            print('Start lines the same!')
+        else:
+            print('Start lines different!')
+
+        # Check data end lines.
+        if len(set(end_line)) == 1:
+            print('End lines the same!')
+        else:
+            print('End lines different!')
+
+        # Make sure data lengths are the same.
+        data_length = ([end_line[i_line] - start_line[i_line] for i_line,
+                        line_num in enumerate(start_line)])
+
+        if len(set(data_length)) == 1:
+            print('Data sizes are the same!')
+        else:
+            print('Data sizes are different!')
+
+        print('Start Line: ', start_line[0])
+        print('End Line: ', end_line[0])
+        print('Original Difference: ', end_line[0] - start_line[0])
+        first_samps = start_line[0]
+        print('New first_samps: ', first_samps)
+        diff = end_line[0] - (start_line[0])
+
+        # Number if rows in data file depends on data file type.
+        if filetype[0] == 'non-parsed':
+            last_samps = ((diff * len(blk_names[0])) // (source_num[0]))
+        elif filetype[0] == 'parsed':
+            last_samps = diff * len(blk_names[0])
+
+        # First sample is technically sample 0, not the start line in the file.
+        first_samps = 0
+
+        print('New last_samps: ', last_samps)
+        print('New Difference: ', last_samps - first_samps)
+
+        super(RawBOXY, self).__init__(
+            info, preload, filenames=[fname], first_samps=[first_samps],
+            last_samps=[last_samps - 1],
+            raw_extras=[raw_extras], verbose=verbose)
+
+    def _read_segment_file(self, data, idx, fi, start, stop, cals, mult):
+        """Read a segment of data from a file.
+
+        Boxy file organises data in two ways, parsed or un-parsed.
+        Regardless of type, output has (n_montages x n_sources x n_detectors
+        + n_marker_channels) rows, and (n_timepoints x n_blocks) columns.
+        """
```

Comment:
nesting the scipy.io import here (within the _read_segment_file function) instead of at the top of the file should address the nested import warning

Reply:
That did the trick, thanks!